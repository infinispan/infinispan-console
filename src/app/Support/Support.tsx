import * as React from 'react';
import {
  Button,
  Modal,
  Stack,
  StackItem,
  Text,
  TextContent,
  TextVariants,
} from '@patternfly/react-core';
import {
  global_FontSize_4xl,
  global_spacer_md,
  global_spacer_sm,
  global_warning_color_100,
} from '@patternfly/react-tokens';
import { ExclamationTriangleIcon } from '@patternfly/react-icons';
import { useTranslation } from 'react-i18next';

/**
 * Support pop up when the user has run the server and there is no user/password to log in
 */
const Support = (props: { isModalOpen: boolean; closeModal: () => void }) => {
  const { t } = useTranslation();
  const brandname = t('brandname.brandname');
  const header = (
    <Stack hasGutter={true}>
      <StackItem>
        <TextContent>
          <Text component={TextVariants.h1}>
            <ExclamationTriangleIcon
              style={{
                color: global_warning_color_100.value,
                fontSize: global_FontSize_4xl.value,
                verticalAlign: 'middle',
                marginRight: global_spacer_md.value,
                marginBottom: global_spacer_sm.value,
              }}
            />
            User not configured
          </Text>
        </TextContent>
      </StackItem>
      <StackItem>
        <TextContent>
          <Text>
            By default, Infinispan Server requires user authentication, but you
            did not configure any user before running the server.
          </Text>
        </TextContent>
      </StackItem>
    </Stack>
  );

  return (
    <Modal
      className="pf-m-redhat-font"
      header={header}
      width={'80%'}
      isOpen={props.isModalOpen}
      onClose={props.closeModal}
      aria-label="Unable to log"
      actions={[
        <Button aria-label={'Reload'} key="reload" onClick={props.closeModal}>
          Reload
        </Button>,
      ]}
    >
      <TextContent>
        <Text component={TextVariants.h6}>Download and run</Text>
        <Text component={TextVariants.pre}>
          ./bin/cli.sh user create admin -p pass
        </Text>
        <Text component={TextVariants.pre}>./bin/server.sh</Text>
        <Text component={TextVariants.h6}>Podman</Text>
        <Text component={TextVariants.pre}>
          podman run --net=host -p 11222:11222 -e USER="admin" -e PASS="pass"
          quay.io/infinispan/server:12.0
        </Text>
        <Text component={TextVariants.h6}>Docker</Text>
        <Text component={TextVariants.pre}>
          docker run -it -p 11222:11222 -e USER="admin" -e PASS="pass"
          infinispan/server:12.0
        </Text>
      </TextContent>
    </Modal>
  );
};

export { Support };
