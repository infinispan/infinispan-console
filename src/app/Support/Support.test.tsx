import {Support} from "@app/Support/Support";
import React from "react";
import {Modal} from "@patternfly/react-core";
import {render} from "@testing-library/react";

describe('Support page', () => {
  test('should render Support component', () => {
    const view = render(<Support isModalOpen={true} closeModal={() => {}}/>);
    expect(view.container.firstChild).toMatchSnapshot();
  });

  // it('should hide the modal when the support page is closed', () => {
  //   const support = render(<Support isModalOpen={false} closeModal={() => {}} />);
  //   let supportModal = support(Modal);
  //   expect(supportModal.props().isOpen).toBe(false);
  // });
  //
  // it('should call the close function when the modal is closed', () => {
  //   let click = 0;
  //   const support = mount(<Support isModalOpen={false} closeModal={() => click=42} />);
  //   let supportModal = support.find(Modal);
  //   expect(supportModal.props().onClose).toBeDefined();
  //   // @ts-ignore
  //   supportModal.props().onClose();
  //   expect(click).toBe(42);
  // });
});
