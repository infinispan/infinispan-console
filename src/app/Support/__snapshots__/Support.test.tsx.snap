// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Support page should render Support component 1`] = `null`;
